lib/buildpack/detect/detecter.rb

```diff
@@ -18,9 +18,14 @@ module AspNetCoreBuildpack
   class Detecter
     def detect(dir)
       fromsource = false
-      arr = Dir.glob(File.join(dir, '**', 'project.json')).map { |file| File.dirname(file) }
-      arr.each do |directory|
-        fromsource = Dir.glob(File.join(directory, '**', '*.??')).grep(/.+\.(?:cs|fs)/).any? unless fromsource
+
+      dirs_to_check = Dir.glob(File.join(dir, '**', 'project.json')).map { |file| File.dirname(file) }
+      dirs_to_check += Dir.glob(File.join(dir, '**', '*.csproj')).map { |file| File.dirname(file) }
```

Comment:
`Dir.glob(File.join(dir, '**', '*.??proj')).grep(/.+\.(?:csproj|fsproj)/).map { |file| File.dirname(file) }` should match both csproj folders and fsproj folders without iterating the directories more than once, and allows us to easily add support for vbproj and other project formats without adding an extra iteration of the directories for each project type added.